Services/WatchlistService.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  WatchlistService.swift
+//  Movies
+//
+//  Created by Diana Chizhik on 21.07.22.
+//
+
+import UIKit
+
+protocol WatchlistServiceProtocol {
+    func getStatus(for id: Int)-> WatchlistStatus?
+    func toggleStatus(for item: WatchlistItem) -> WatchlistStatus?
+    func getWatchlistItems() -> [WatchlistItem]?
+}
+
+enum WatchlistStatus {
+    case added, notAdded
+}
+
+class WatchlistService {
+    typealias Watchlist = [Int : WatchlistItem]
+    
+    func getStatus(for id: Int)-> WatchlistStatus? {
+        if let watchlist = getWatchlist() {
+            return watchlist[id] != nil ? .added : .notAdded
+        } else {
+            return nil
+        }
+    }
```

Comment:
What if we make `getWatchlist` return an empty dictionary in case nothing is stored instead of `nil`? Then this method can be a lot more simpler and we get rid of the `nil` status which seems weird. If `.notAdded` means that is not added yet, what does `nil` means?

```suggestion
    func getStatus(for id: Int)-> WatchlistStatus {
        let watchlist = getWatchlist()
        return watchlist[id] != nil ? .added : .notAdded
    }
```